### PR TITLE
[serving] set default `vllm_seed` param for `LocalModelLLMServing_vllm` to `None` to avoid warning

### PR DESCRIPTION
**Pull request by SunnyHaze**
-  Avoid `Warning: FlashInfer 0.2.3+ does not support per-request generators. Falling back to PyTorch-native implementation.` when using VLLM>=0.9.2 as backend.
- Related Links:
  - https://github.com/vllm-project/vllm/issues/17898
  - https://chatgpt.com/c/68786709-8100-800c-8d44-28437523da0d
  - https://github.com/flashinfer-ai/flashinfer/issues/1104

### CODE REVIEW

**Comment by SunnyHaze** [comment]
<html>
<body>
<html><head></head><body><p>你贴出的代码看起来确实<strong>没有显式使用 per-request generator</strong> —— 你为所有请求统一传入了同一个 <code inline="">SamplingParams</code> 实例，这本应避免触发 FlashInfer 的 fallback。但仍然触发了这条日志：</p>
<pre><code>FlashInfer 0.2.3+ does not support per-request generators. Falling back to PyTorch-native implementation.
</code></pre>
<p>这说明还有<strong>其他可能原因</strong>导致 fallback：</p>
<hr>
<h3>✅ 真正触发 fallback 的代码路径是：</h3>
<pre><code class="language-python">if generators:
    logger.warning("FlashInfer 0.2.3+ does not support per-request generators. Falling back to PyTorch-native implementation.")
    return self.forward_native(logits, generators, k, p)
</code></pre>
<p>其中 <code inline="">generators</code> 是一个 <code inline="">dict[int, torch.Generator]</code>，只要这个 <code inline="">dict</code> 非空，就会触发 fallback。</p>
<hr>
<h3>🔍 所以根因不是 sampling param，而是：</h3>
<p>你在某个地方（或 vLLM 内部）传入了 <code inline="">per-request generator</code>：</p>
<pre><code class="language-python">generators = {0: torch.Generator(device="cuda").manual_seed(42), ...}
</code></pre>
<blockquote>
<p>**vLLM 会自动为每个 request 创建 generator（基于 seed）来保证可重复性。**只要你传了 <code inline="">seed</code>（哪怕统一），vLLM 就会在内部构建 <code inline="">per-request generator</code> 字典（每个请求一个 Generator），最终这个 <code inline="">generators</code> 参数非空，从而 fallback。</p>
</blockquote>
<hr>
<h3>🧠 所以，你有两个选择：</h3>
<h4>✅ 方案一：接受 fallback（仍然能跑，只是慢一点）</h4>
<p>你已经设置了 <code inline="">seed=42</code>，vLLM 会构建 <code inline="">generators</code>，FlashInfer 目前无法支持这个路径，因此 fallback 是预期行为。<br>
<strong>你无需担心准确性，只是牺牲了一点速度。</strong></p>
<h4>✅ 方案二：强制启用 FlashInfer（放弃 determinism）</h4>
<p>如果你非要用 FlashInfer 并 <strong>强制启用加速（避免 fallback）</strong>，你需要：</p>
<ol>
<li>
<p><strong>不传 seed</strong>（让 vLLM 不创建 <code inline="">generator</code>）；</p>
</li>
<li>
<p>或者 <strong>修改 vLLM 源码中 fallback 的条件判断逻辑（不推荐）</strong>。</p>
</li>
</ol>
<p>你可以试试把这行删掉：</p>
<pre><code class="language-python">seed=vllm_seed  # ← 注释掉这个
</code></pre>
<p>即：</p>
<pre><code class="language-python">self.sampling_params = SamplingParams(
    temperature=vllm_temperature,
    top_p=vllm_top_p,
    max_tokens=vllm_max_tokens,
    top_k=vllm_top_k,
    repetition_penalty=vllm_repetition_penalty,
    # seed=vllm_seed ← 不传就不会创建 per-request generator
)
</code></pre>
<p>⚠️ 这样就失去了可重复性，但 FlashInfer 加速就能启用了。</p>
<hr>
<h3>📌 总结</h3>

选项 | 是否传 seed | 是否使用 FlashInfer | 是否可复现输出 | 性能
-- | -- | -- | -- | --
✅ 推荐默认 | ✅传 seed | ❌ 否（会 fallback） | ✅ 可复现 | 中等
🧪 若要加速 | ❌不传 seed | ✅ 是 | ❌ 不可复现 | 更快


<hr>
<p>如果你愿意，我可以帮你 patch 一段代码，在不传 seed 时自动判断是否启用 FlashInfer。是否需要？</p></body></html>
</body>
</html>
